servant-client/src/Servant/Client.hs

```diff
@@ -60,6 +62,24 @@ import           Servant.Common.Req
 client :: HasClient layout => Proxy layout -> Client layout
 client p = clientWithRoute p defReq
 
+-- | 'staticClient' allows you to provide a static argument to given 'Client layout'
+--
+-- > type MyApi = "books" :> Get '[JSON] [Book] -- GET /books
+-- >         :<|> "books" :> ReqBody '[JSON] Book :> Post '[JSON] Book -- POST /books
+-- >
+-- > myApi :: Proxy MyApi
+-- > myApi = Proxy
+-- >
+-- > getAllBooks :: Manager -> BaseUrl -> ClientM [Book]
+-- > postNewBook :: Book -> Manager -> BaseUrl -> ClientM Book
+-- > (getAllBooks :<|> postNewBook) = client myApi
+-- >
+-- > getAllBooks' :: Manager -> ClientM [Book]
+-- > postNewBook' :: Book -> Manager -> ClientM Book
+-- > (getAllBooks' :<|> postNewBook') = staticClient (BaseUrl Http "localhost" 8080 "/") myApi
+staticClient :: (Enter (Client layout) BaseUrl ret, HasClient layout) => BaseUrl -> Proxy layout -> ret
+staticClient b = enter b . client
```

Comment:
I'm not sure if `staticClient` is really the best name. Does anyone have better ideas?